client/components/tinymce/plugins/media/plugin.jsx

```diff
@@ -145,10 +208,19 @@ function mediaButton( editor ) {
 			return;
 		}
 
+		const dirtyImage = MediaStore.getDirtyImage();
+		let numOfImagesToUpdate = 0;
+
 		isVisualEditMode = ! editor.isHidden();
 
 		if ( isVisualEditMode ) {
```

Comment:
I dont know if we care about this case or not, but when I:
1. Edit image
2. Click done
3. Click outside image gallery
4. Click on HTML view

I end up in html view before callback kicks in, Im in html mode now and I have old image in the source.
It's actually easy to do on the slow wifi Im on